advisor_engine/config.py

```diff
@@ -37,66 +37,115 @@
-            'propagate': False
+FILE_LOGGING = os.environ.get('FILE_LOGGING', '').lower() == 'true'
+
+# Toggle file logging or stdout only
+if FILE_LOGGING:
+    os.makedirs(LOG_DIR, exist_ok=True) # Ensure logging directory exists
+    LOGGING_CONFIG = { 
+        'version': 1,
+        'disable_existing_loggers': True,
+        'formatters': {
+            'simple': {
+                'format': '<%(asctime)s> - %(levelname)s - %(message)s'
+            },
+            'logstash': { 
+                'class': 'logstash_formatter.LogstashFormatter'
+            },
         },
-        'api': {
-            'handlers': ['stdout', 'api_file'],
-            'level': API_LOG_LEVEL,
-            'propagate': False
+        'handlers': { 
+            'stdout': { 
+                'level': STDOUT_LOG_LEVEL,
+                'formatter': 'simple' if SIMPLE_LOGS else 'logstash',
+                'class': 'logging.StreamHandler'
+            },
+            'engine_file': {
+                'level': ENGINE_LOG_LEVEL,
+                'formatter': 'simple' if SIMPLE_LOGS else 'logstash',
+                'class': 'logging.handlers.RotatingFileHandler',
+                'filename': ENGINE_LOG_FILE,
+                'maxBytes': ENGINE_MAX_BYTES,
+                'backupCount': ENGINE_BACKUP_COUNT
+            },
+            'api_file': {
+                'level': API_LOG_LEVEL,
+                'formatter': 'simple' if SIMPLE_LOGS else 'logstash',
+                'class': 'logging.handlers.RotatingFileHandler',
+                'filename': API_LOG_FILE,
+                'maxBytes': API_MAX_BYTES,
+                'backupCount': API_BACKUP_COUNT
+            }
         },
-        'uvicorn': {
-            'handlers': ['stdout', 'api_file'],
-            'level': API_LOG_LEVEL,
-            'propagate': False
+        'loggers': {
+            'engine': {
+                'handlers': ['stdout', 'engine_file'],
+                'level': ENGINE_LOG_LEVEL,
+                'propagate': False
+            },
+            'api': {
+                'handlers': ['stdout', 'api_file'],
+                'level': API_LOG_LEVEL,
+                'propagate': False
+            },
+            'uvicorn': {
+                'handlers': ['stdout', 'api_file'],
+                'level': API_LOG_LEVEL,
+                'propagate': False
+            },
+            'uvicorn.access': {
+                'handlers': ['stdout', 'api_file'],
+                'level': API_LOG_LEVEL,
+                'propagate': False
+            },
+            'uvicorn.error': {
+                'handlers': ['stdout', 'api_file'],
+                'level': API_LOG_LEVEL,
+                'propagate': False
+            }
+        } 
+    }
+else:
+    LOGGING_CONFIG = { 
+        'version': 1,
+        'disable_existing_loggers': True,
+        'formatters': {
+            'simple': {
+                'format': '<%(levelname)s> - %(asctime)s - %(message)s'
```

Comment:
According to https://0pointer.de/blog/projects/journal-submit.html / http://0pointer.de/public/systemd-man/systemd.exec.html / http://0pointer.de/public/systemd-man/sd-daemon.html this needs to be the level "number", not name?

Reply:
And the numbers in Python (via `levelno`) are different: https://docs.python.org/3/library/logging.html#logging-levels

But as said on Slack, I don't think we need the systemd specific tagging here, so if it just shows up as a string is probably just fine

Reply:
I have changed `levelname` to `levelno`. If we don't want those in the log output I can remove it entirely.